core/meta/src/TClass.cxx

```diff
@@ -5464,29 +5464,24 @@ TClass *TClass::LoadClassCustom(const char *requestedname, Bool_t silent)
 
 void TClass::LoadClassInfo() const
 {
-   R__LOCKGUARD(gInterpreterMutex);
+   // Load class info if not already loaded and auto-parsing is not suspended
+   if (!fClassInfo && !gInterpreter->IsAutoParsingSuspended()) {
```

Comment:
I think you have a good point in that case we can simplify it to just
```
if (R__likely(!fCanLoadClassInfo)) return; // fClassInfo already loaded or not loadable.
R___LOCKGUARD(gROOTMutext);
if (!fCanLoadClassInfo || fClassInfo) return; // in case some other thread beat us to the punch and we are already done.
...
// Previous version of the code
```